src/tempdir.rs

```diff
@@ -54,30 +50,49 @@ impl TempDir {
         }
         dir_bytes.pop(); // Remove the null becasue from_vec can't handle it.
         Ok(TempDir {
-            path: Some(PathBuf::from(OsString::from_vec(dir_bytes))),
+            path: PathBuf::from(OsString::from_vec(dir_bytes)),
         })
     }
 
-    /// Removes the temporary directory.  Calling this is optional as dropping a `TempDir` object
-    /// will also remove the directory.  Calling remove explicitly allows for better error handling.
-    pub fn remove(mut self) -> Result<()> {
-        let path = self.path.take();
-        path.map_or(Ok(()), fs::remove_dir_all)?;
-        Ok(())
+    /// Removes the temporary directory.
+    ///
+    /// Calling this is optional as when a `TempDir` object goes out of scope,
+    /// the directory will be removed.
+    /// Calling remove explicitly allows for better error handling.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use std::path::Path;
+    /// # use std::path::PathBuf;
+    /// # use vmm_sys_util::tempdir::TempDir;
+    /// let temp_dir = TempDir::new("/tmp/testdir").unwrap();
+    /// temp_dir.remove().unwrap();
+    ///
+    pub fn remove(&self) -> Result<()> {
```

Comment:
Should we use some marker to store the state whether the tempdir has already been deleted to avoid duplicated deleting?

Reply:
I think the function will return error in the case where remove is called twice. I would rather add a documentation entry specifying it should only be called once.